### ISSUE TITLE
[Composer\Downloader\TransportException] Operation timed out

### ISSUE THREAD

**Issue by vicguedez** [composer/composer#5154]
Hello,

I'm new to Debian 8, trying to install 'facebook\xhp-lib'. I've got a 10Mbps connection and can navigate at full speed, but when using composer it just gets timeout.

I'm using the following composer.json:

```json
{
    "config": {
      "secure-http": false,
      "optimize-autoloader": true
    },
    "require": {
      "facebook/xhp-lib": "2.3"
    }
}
```

When I run this command:

```
composer install -vvv
```

I get this output:

```
Reading ./composer.json
Loading config file /home/victor/.composer/config.json
Loading config file /home/victor/.composer/auth.json
Loading config file ./composer.json
Checking CA file /etc/ssl/certs/ca-certificates.crt
Executing command (/home/victor/www/app): git branch --no-color --no-abbrev -v
Executing command (/home/victor/www/app): git describe --exact-match --tags
Failed to initialize global composer: Composer could not find the config file: /home/victor/.composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Loading composer repositories with package information
Downloading https://packagist.org/packages.json
SlowTimer [7972ms] at curl: https://packagist.org/packages.json
Writing /home/victor/.cache/composer/repo/https---packagist.org/packages.json into cache
Updating dependencies (including require-dev)
Reading /home/victor/.cache/composer/repo/https---packagist.org/p-provider-2013.json from cache
Downloading http://packagist.org/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json
SlowTimer [60001ms] at curl: http://packagist.org/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json
The "http://packagist.org/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json" file could not be downloaded: Failed to open http://packagist.org/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json (Operation timed out after 59860 milliseconds with 474681 out of 857622 bytes received)
Retrying with degraded mode, check https://getcomposer.org/doc/articles/troubleshooting.md#degraded-mode for more info
Downloading http://packagist.org/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json
SlowTimer [60001ms] at curl: http://87.98.253.214/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json
Downloading http://packagist.org/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json
SlowTimer [59999ms] at curl: http://87.98.253.214/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json
Downloading http://packagist.org/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json
SlowTimer [60000ms] at curl: http://87.98.253.214/p/provider-2014%246e256ac757a31949f2560c57b83ff57649232e226864f330c6556e4134ee5d52.json

                                                                               
  [Composer\Downloader\TransportException]                                     
  The "http://packagist.org/p/provider-2014%246e256ac757a31949f2560c57b83ff57  
  649232e226864f330c6556e4134ee5d52.json" file could not be downloaded: Faile  
  d to open http://87.98.253.214/p/provider-2014%246e256ac757a31949f2560c57b8  
  3ff57649232e226864f330c6556e4134ee5d52.json (Operation timed out after 5957  
  4 milliseconds with 358836 out of 857622 bytes received)                     
                                                                               

Exception trace:
 () at phar://composer.phar/bin/../src/../src/Composer/Util/RemoteFilesystem.php:452
 Composer\Util\RemoteFilesystem->get() at phar://composer.phar/bin/../src/../src/Composer/Util/RemoteFilesystem.php:99
 Composer\Util\RemoteFilesystem->getContents() at phar://composer.phar/bin/../src/../src/Composer/Repository/ComposerRepository.php:644
 Composer\Repository\ComposerRepository->fetchFile() at phar://composer.phar/bin/../src/../src/Composer/Repository/ComposerRepository.php:557
 Composer\Repository\ComposerRepository->loadProviderListings() at phar://composer.phar/bin/../src/../src/Composer/Repository/ComposerRepository.php:288
 Composer\Repository\ComposerRepository->whatProvides() at phar://composer.phar/bin/../src/../src/Composer/DependencyResolver/Pool.php:204
 Composer\DependencyResolver\Pool->computeWhatProvides() at phar://composer.phar/bin/../src/../src/Composer/DependencyResolver/Pool.php:193
 Composer\DependencyResolver\Pool->whatProvides() at phar://composer.phar/bin/../src/../src/Composer/DependencyResolver/RuleSetGenerator.php:168
 Composer\DependencyResolver\RuleSetGenerator->whitelistFromPackage() at phar://composer.phar/bin/../src/../src/Composer/DependencyResolver/RuleSetGenerator.php:322
 Composer\DependencyResolver\RuleSetGenerator->getRulesFor() at phar://composer.phar/bin/../src/../src/Composer/DependencyResolver/Solver.php:214
 Composer\DependencyResolver\Solver->solve() at phar://composer.phar/bin/../src/../src/Composer/Installer.php:498
 Composer\Installer->doInstall() at phar://composer.phar/bin/../src/../src/Composer/Installer.php:228
 Composer\Installer->run() at phar://composer.phar/bin/../src/../src/Composer/Command/InstallCommand.php:134
 Composer\Command\InstallCommand->execute() at phar://composer.phar/bin/../src/../vendor/symfony/console/Command/Command.php:259
 Symfony\Component\Console\Command\Command->run() at phar://composer.phar/bin/../src/../vendor/symfony/console/Application.php:844
 Symfony\Component\Console\Application->doRunCommand() at phar://composer.phar/bin/../src/../vendor/symfony/console/Application.php:192
 Symfony\Component\Console\Application->doRun() at phar://composer.phar/bin/../src/../src/Composer/Console/Application.php:181
 Composer\Console\Application->doRun() at phar://composer.phar/bin/../src/../vendor/symfony/console/Application.php:123
 Symfony\Component\Console\Application->run() at phar://composer.phar/bin/../src/../src/Composer/Console/Application.php:99
 Composer\Console\Application->run() at phar://composer.phar/bin/composer:43
 include() at /usr/local/bin/composer:24

install [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--no-plugins] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--ignore-platform-reqs] [--] [<packages>]...
```

I have read and tried lot of things (process_timeout, running with hhvm, forcing to https, etc...), and nothing have worked... Is there any workaround this?

Thanks anyway.

PS: Sorry for bad English.

**Comment by alcohol** [comment]
I'm sorry but this looks like a networking issue to me. There is very little Composer can do about that. It is also not easy to debug remotely, so I am not sure how we can help you.

**Comment by vicguedez** [comment]
Well, I do not know if it was a network issue, I know it seems like it. I got this to work by running composer through hhvm and forcing SocketTimeout and SlowQueryThreshold like so:

Add this alias to ~/.bashrc with `sudo nano ~/.bashrc`:

>alias composer="hhvm -v ResourceLimit.SocketDefaultTimeout=180 -v Http.SlowQueryThreshold=300000 /usr/local/bin/composer"

Run `source ~/.bashrc`, and then just run composer for your project.

I hope that helps someone.

PS: The command didn't take that long I just set those high values just in case.
